consumer.go

```diff
@@ -97,7 +107,7 @@ var ErrSkipCheckpoint = errors.New("skip checkpoint")
 // Scan launches a goroutine to process each of the shards in the stream. The ScanFunc
 // is passed through to each of the goroutines and called with each message pulled from
 // the stream.
-func (c *Consumer) Scan(ctx context.Context, fn ScanFunc) error {
+func (c *Consumer) Scan(ctx context.Context, fn interface{}) error {
```

Comment:
I think we should create another method rather than changing the type to a naked interface. Creating a separate method with explicit types helps catch things at the compilation stage rather than risking a runtime error.

Reply:
c3365d0

Reply:
I see you linked to a new commit but I don't see `ScanBatch` defined there.

Reply:
5c2e393